libraries/lib-wave-track/WaveTrack.h

```diff
@@ -1058,6 +1058,10 @@ class WAVE_TRACK_API WaveTrack final
       void ConvertToSampleFormat(sampleFormat format,
          const std::function<void(size_t)> & progressReport = {});
 
+      //! Silences the 'length' amount of samples starting from 'offset'
```

Comment:
Seems to me like `start` is actually treated as an absolute time in this method's body.

Reply:
```
void WaveClip::SetSilence(sampleCount offset, sampleCount length)
{
   const auto start = TimeToSamples(mTrimLeft) + offset;
   Transaction transaction{ *this };
   for (auto &pSequence : mSequences)
      pSequence->SetSilence(start, length);
   transaction.Commit();
   MarkChanged();
}
```

I don't think so?

`start` is a count of samples given to the `Sequence` object, which doesn't know the timeline.  It doesn't even know the rate.  `WaveClip` knows its rate and offset.

Reply:
Right - I was confusing `WaveTrack::Silence` and `WaveTrack::Interval::SetSilence`: the former takes an absolute value - which makes sense - and makes it relative to clip play start.